internal/api/grpc/server/gateway.go

```diff
@@ -274,7 +275,11 @@ func grpcCredentials(tlsConfig *tls.Config) credentials.TransportCredentials {
 func setRequestURIPattern(ctx context.Context) {
 	pattern, ok := runtime.HTTPPathPattern(ctx)
 	if !ok {
-		return
+		// As all unmatched paths will be handled by the gateway, any request not matching a pattern,
+		// means there's no route to the path.
+		// To prevent high cardinality on metrics and tracing, we want to make sure we don't record
+		// the actual path as name (it will still be recorded explicitly in the span http info). 
+		pattern = UnknownPath
 	}
 	span := trace.SpanFromContext(ctx)
```

Comment:
would it make sense to add the unknown pattern to the span so that we see which urls people call?

Reply:
We get the actual path as http.target on the trace, but also the trace name will be `UNKNOWN_PATH`, which allows grouping them easier and reduces high cardinality there as well.
You got a DM with a link to my tests.

Reply:
![image](https://github.com/user-attachments/assets/f377cf7c-86bf-41bc-b7d9-4a07053aa723)

Reply:
Ah nice my fault